Increase dynamicDowncast<> usage in CachedResourceLoader

annevk opened the pull request:
#### 956dd02e69b40151b55f7fc3bc81489c42894211
<pre>
Increase dynamicDowncast&lt;&gt; usage in CachedResourceLoader
<a href="https://bugs.webkit.org/show_bug.cgi?id=270049">https://bugs.webkit.org/show_bug.cgi?id=270049</a>

Reviewed by Chris Dumez.

Minor change to avoid some redundant downcasting. Also remove
hasSVGImage() from CachedImage as you can use is&lt;SVGImage&gt; on a
potentially null object.

* Source/WebCore/loader/cache/CachedImage.cpp:
(WebCore::CachedImage::hasSVGImage const): Deleted.
* Source/WebCore/loader/cache/CachedImage.h:
* Source/WebCore/loader/cache/CachedResourceLoader.cpp:
(WebCore::isSVGImageCachedResource):
(WebCore::cachedResourceSVGImage):

Canonical link: <a href="https://commits.webkit.org/275334@main">https://commits.webkit.org/275334@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f2549a3e0c92fbf23776683229c9eb26423f5ef9

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/41539 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/20553 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/43917 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/44107 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/37631 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/43846 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/23661 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/17883 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/34348 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/42113 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/17481 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/35784 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/15009 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/15197 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/36790 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/45477 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/37727 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/37110 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/40867 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/16354 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/13434 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/39275 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/17973 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9309 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/18029 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/17617 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/loader/cache/CachedResourceLoader.cpp:
```diff
@@ -987,14 +987,15 @@ static FetchOptions::Destination destinationForType(CachedResource::Type type, L
+    if (!cachedImage)
         return nullptr;
-    return downcast<SVGImage>(downcast<CachedImage>(*resource).image());
+    return dynamicDowncast<SVGImage>(cachedImage->image());
```
nit: could have used a ternary:
```
return cachedImage ? dynamicDowncast<SVGImage>(cachedImage->image()) : nullptr;
```

annevk committed:
Increase dynamicDowncast<> usage in CachedResourceLoader
https://bugs.webkit.org/show_bug.cgi?id=270049

Reviewed by Chris Dumez.

Minor change to avoid some redundant downcasting. Also remove
hasSVGImage() from CachedImage as you can use is<SVGImage> on a
potentially null object.

* Source/WebCore/loader/cache/CachedImage.cpp:
(WebCore::CachedImage::hasSVGImage const): Deleted.
* Source/WebCore/loader/cache/CachedImage.h:
* Source/WebCore/loader/cache/CachedResourceLoader.cpp:
(WebCore::isSVGImageCachedResource):
(WebCore::cachedResourceSVGImage):

Canonical link: https://commits.webkit.org/275334@main
